docs/site/Sequence.md

```diff
@@ -336,3 +336,90 @@ An example error message when the debug mode is enabled:
 - Try and use existing actions
 - Implement your own version of built in actions
 - Publish reusable actions to npm
+
+## Working with Express middleware
+
+Under the hood, LoopBack leverages [Express](https://expressjs.com) framework
+and its concept of middleware. To avoid common pitfalls, it is not possible to
+mount Express middleware directly on a LoopBack application. Instead, LoopBack
+provides and enforces a higher-level structure.
+
+In a typical Express application, there are four kinds of middleware invoked in
+the following order:
+
+1. Request-preprocessing middleware like
+   [cors](https://www.npmjs.com/package/cors) or
+   [body-parser](https://www.npmjs.com/package/body-parser).
+2. Route handlers handling requests and producing responses.
+3. Middleware serving static assets (files).
+4. Error handling middleware.
+
+In LoopBack, we handle the request in the following steps:
+
+1. The built-in request-preprocessing middleware is invoked.
+2. The registered Sequence is started. The default implementation of `findRoute`
+   and `invoke` actions will try to match the incoming request against the
+   following resources:
+   1. Native LoopBack routes (controller methods, route handlers).
+   2. External Express routes (registered via `mountExpressRouter` API)
```

Comment:
Does it mean that we cannot mount an express middleware to be invoked before controller methods?

Reply:
Yes, exactly. You have to mount LB4 on Express app if you want to run custom express middleware before controller methods, I am explaining this in more details later in the text.

Reply:
Reworded as follows:

```markdown
In LoopBack, we use async functions instead of callbacks and thus can use simple
`try`/`catch` flow to receive both sync and async errors from individual
sequence actions. A typical Sequence implementation then passes these errors to
the Sequence action `reject`.

You can learn more about error handling in
[Handling errors](Sequence.md#handling-errors).
```